src/iosource/Manager.cc

```diff
@@ -76,6 +76,9 @@ Manager::~Manager() {
 
     pkt_dumpers.clear();
 
+    // Was registered without lifetime management.
+    delete pkt_src;
```

Comment:
It would be reaped too early, yes. The loop queries the pkt_src to determine whether updating network time is okay. If the packet source vanishes, that (currently) implies using realtime for entwork time. Matters mostly for pcaps read with `exit_only_after_terminate=T`.

It's in a separate commit, so hope someone git annotating has enough context for a later rework.